src/hotspot/os/windows/os_windows.cpp

```diff
@@ -2909,20 +2909,23 @@ LONG WINAPI topLevelVectoredExceptionFilter(struct _EXCEPTION_POINTERS* exceptio
 
 #if defined(USE_VECTORED_EXCEPTION_HANDLING)
 LONG WINAPI topLevelUnhandledExceptionFilter(struct _EXCEPTION_POINTERS* exceptionInfo) {
-  if (InterceptOSException) goto exit;
-  DWORD exception_code = exceptionInfo->ExceptionRecord->ExceptionCode;
+  if (InterceptOSException) {
```

Comment:
I may be missing something, but why not just use `if (!InterceptOSException)` here and get rid of the `goto`?

Reply:
This decision was based on a review comment in https://github.com/openjdk/jdk/pull/15096, which preferred keeping InterceptOSException without the negation

Reply:
Okay. If an early bailout is desirable, I suggest instead putting everything in the `else` block into a helper method, and then have `if (InterceptOSException) return;` at the start. `topLevelUnhandledExceptionFilter` can then do:

```
helper(exceptionInfo); // or whatever name is chosen
return previousUnhandledExceptionFilter ? previousUnhandledExceptionFilter(exceptionInfo) : EXCEPTION_CONTINUE_SEARCH;
```

Reply:
Hmm, creating an entire method exclusively for this seems a bit excessive, what if I just surrounded the else block with a scope and kept everything else the same? The issue is goto jumping across the locals of this method after all, which a scope coould solve

Reply:
That seems fine

Reply:
Daniel suggested splitting initialization into declaration and assignment instead, which seems like a better solution

Reply:
Okay. TBH, I don't see how this is an improvement over the status quo, since the `goto` still jumps over the initialization...

Reply:
The little gotcha in there is that goto is allowed to jump over the declaration of a scalar (of primitive type) local, but not the definition of one (declaration + initialization together). Hence goto is not allowed to jump over address pc = x; for instance, but is allowed to for address pc; pc = x;

Reply:
Right, so we're employing a workaround to silence the warning. That seems like the tail wagging the dog though. There's a potential issue in this code that the warning tries to point out: the code after the label might accesses the uninitialized variables, since they are in scope. Re-writing the code in the way you did makes the warning go away, but not the potential issue?

Reply:
Not entirely sure what the issue is, since the code that goto jumps to never uses the uninitialized locals. I don't really have any other solutions to solve this issue unfortunately :/ Maybe @djelinski can add some perspective on this?

Reply:
I think killing the goto makes most sense here.

Reply:
I've removed the goto, but would like to wait until Thomas is back

Reply:
Sure, remove the empty block.